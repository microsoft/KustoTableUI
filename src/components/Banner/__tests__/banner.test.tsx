import renderer, { ReactTestRenderer } from 'react-test-renderer';
import { Banner } from '..';
import StarIcon from '@mui/icons-material/Star';

describe('Banner Snapshot UI Tests', () => {
  const testMessage = <span>Show an element</span>;
  const testIcon = <StarIcon fontSize="inherit" sx={{ mr: -1 }} />;
  let component: ReactTestRenderer;

  afterEach(() => component.unmount());

  test.each`
    icon         | message                                           | severity     | variant       | showButton   | isVisible    | scenarioName
    ${undefined} | ${'This is a test message.'}                      | ${'success'} | ${'standard'} | ${true}      | ${true}      | ${'undefined Icon'}
    ${testIcon}  | ${testMessage}                                    | ${'info'}    | ${'standard'} | ${false}     | ${true}      | ${'React Nodes for Icon and Message'}
    ${undefined} | ${'All the props are set to null for this test.'} | ${undefined} | ${undefined}  | ${undefined} | ${undefined} | ${'All Props default'}
  `(
    'Test Banner Display $scenarioName',
    ({ icon, message, severity, variant, showButton, isVisible }) => {
      const testProps: any = {
        icon,
        isVisible,
        message,
        severity,
        showButton,
        variant,
      };
      component = renderer.create(<Banner {...testProps} />);
      const banner = component.toJSON();
      expect(banner).toMatchSnapshot();
    },
  );
});
