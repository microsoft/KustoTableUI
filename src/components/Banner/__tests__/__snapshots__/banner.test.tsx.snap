// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner Snapshot UI Tests Test Banner Display All Props default 1`] = `null`;

exports[`Banner Snapshot UI Tests Test Banner Display React Nodes for Icon and Message 1`] = `
<div
  className=""
  style={
    Object {
      "display": "block",
    }
  }
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardInfo MuiAlert-standard css-1fuz9qr-MuiPaper-root-MuiAlert-root"
    role="alert"
  >
    <div
      className="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-s54opt-MuiSvgIcon-root"
        data-testid="StarIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
    </div>
    <div
      className="MuiAlert-message css-1pxa9xg-MuiAlert-message"
    >
      <span>
        Show an element
      </span>
    </div>
  </div>
</div>
`;

exports[`Banner Snapshot UI Tests Test Banner Display undefined Icon 1`] = `
<div
  className=""
  style={
    Object {
      "display": "block",
    }
  }
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardSuccess MuiAlert-standard css-wnkddu-MuiPaper-root-MuiAlert-root"
    role="alert"
  >
    <div
      className="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
        data-testid="SuccessOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20,12A8,8 0 0,1 12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4C12.76,4 13.5,4.11 14.2, 4.31L15.77,2.74C14.61,2.26 13.34,2 12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0, 0 22,12M7.91,10.08L6.5,11.5L11,16L21,6L19.59,4.58L11,13.17L7.91,10.08Z"
        />
      </svg>
    </div>
    <div
      className="MuiAlert-message css-1pxa9xg-MuiAlert-message"
    >
      This is a test message.
    </div>
    <div
      className="MuiAlert-action css-ki1hdl-MuiAlert-action"
    >
      <button
        aria-label="Close"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall css-1e0d89p-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        title="Close"
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
          data-testid="CloseIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
